pulsar-broker/src/test/java/org/apache/pulsar/broker/resourcegroup/ResourceGroupRateLimiterTest.java

```diff
@@ -0,0 +1,149 @@
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pulsar.broker.resourcegroup;
+
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertNotNull;
+import static org.testng.Assert.assertNull;
+import org.apache.pulsar.broker.service.BrokerTestBase;
+import org.apache.pulsar.client.admin.PulsarAdminException;
+import org.apache.pulsar.client.api.MessageId;
+import org.apache.pulsar.client.api.Producer;
+import org.apache.pulsar.client.api.PulsarClientException;
+import org.awaitility.Awaitility;
+import org.testng.Assert;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+public class ResourceGroupRateLimiterTest extends BrokerTestBase {
+
+    final String rgName = "testRG";
+    org.apache.pulsar.common.policies.data.ResourceGroup testAddRg =
+    new org.apache.pulsar.common.policies.data.ResourceGroup();
+    final String namespaceName = "prop/ns-abc";
+    final String topicString = "persistent://prop/ns-abc/test-topic";
+
+    @BeforeClass
+    @Override
+    protected void setup() throws Exception {
+        conf.setMaxPendingPublishRequestsPerConnection(0);
+        super.baseSetup();
+        prepareData();
+
+    }
+
+    @AfterClass(alwaysRun = true)
+    @Override
+    protected void cleanup() throws Exception {
+        super.internalCleanup();
+    }
+
+    public void createResourceGroup(String rgName, org.apache.pulsar.common.policies.data.ResourceGroup rg) throws PulsarAdminException {
+        admin.resourcegroups().createResourceGroup(rgName, rg);
+
+        Awaitility.await().untilAsserted(() -> {
+            final org.apache.pulsar.broker.resourcegroup.ResourceGroup resourceGroup = pulsar
+                .getResourceGroupServiceManager().resourceGroupGet(rgName);
+            assertNotNull(resourceGroup);
+            assertEquals(rgName, resourceGroup.resourceGroupName);
+        });
+
+    }
+
+    public void deleteResourceGroup(String rgName) throws PulsarAdminException {
+        admin.resourcegroups().deleteResourceGroup(rgName);
+        Awaitility.await().atMost(1, TimeUnit.SECONDS)
+            .untilAsserted(() -> assertNull(pulsar.getResourceGroupServiceManager().resourceGroupGet(rgName)));
+    }
+
+    @Test
+    public void testResourceGroupPublishRateLimit() throws Exception {
+
+        createResourceGroup(rgName, testAddRg);
```

Comment:
What is the ResourceGroup rate limiting set to?  I don't see any explicit setting of any limits.

Reply:
it is set to 1 msg/second, 10bytes/second in prepareData() method.